components/client/src/test/integration/scala/com/hotels/styx/client/RetryHandlingSpec.scala

```diff
@@ -134,25 +136,37 @@ class RetryHandlingSpec extends FunSuite with BeforeAndAfterAll with Matchers wi
     val response = waitForResponse(client.sendRequest(request))
   }
 
+  def activeOrigins(backendService: BackendService): ActiveOrigins = newOriginsInventoryBuilder(backendService).build()
+
+  def roundRobinStrategy(activeOrigins: ActiveOrigins): RoundRobinStrategy = new RoundRobinStrategy(activeOrigins)
+
+  def stickySessionStrategy(activeOrigins: ActiveOrigins) = new StickySessionLoadBalancingStrategy(activeOrigins, roundRobinStrategy(activeOrigins))
+
+
   test("retries the next available origin on failure") {
-    val client: StyxHttpClient = newHttpClientBuilder(new BackendService.Builder()
+    val backendService = new BackendService.Builder()
       .origins(unhealthyOriginOne, unhealthyOriginTwo, unhealthyOriginThree, healthyOriginTwo)
-      .build())
-      .retryPolicy(new RetryNTimes(3))
+      .build()
+    val client: StyxHttpClient = newHttpClientBuilder(backendService)
+      .retryPolicy(new RetryNTimes(5))
+      .loadBalancingStrategy(stickySessionStrategy(activeOrigins(backendService)))
       .build
 
     val request: HttpRequest = get("/version.txt").build
 
     val response = waitForResponse(client.sendRequest(request))
 
+    println("response: " + response)
+
     assertThat(response.header(ORIGIN_ID_DEFAULT).get(), containsString("HEALTHY_ORIGIN_TWO"))
   }
 
   test("propagates the last observed exception if all retries failed") {
-    val client: StyxHttpClient = newHttpClientBuilder(
-      new BackendService.Builder()
-        .origins(unhealthyOriginOne, unhealthyOriginTwo, unhealthyOriginThree)
-        .build())
+    val backendService = new BackendService.Builder()
+      .origins(unhealthyOriginOne, unhealthyOriginTwo, unhealthyOriginThree)
+      .build()
+    val client: StyxHttpClient = newHttpClientBuilder(backendService)
+      .loadBalancingStrategy(stickySessionStrategy(activeOrigins(backendService)))
```

Comment:
Why are we using sticky-sessions here? This seems to be mixing unrelated features into the test.

Reply:
In the original tests the client was configured to use sticky sessions. Now this behaviour is injected in in a form of load balancing strategy. I have simply kept the client configuration as it was previously.